app/models/user.rb

```diff
@@ -36,6 +36,9 @@ class User < ApplicationRecord
   scope :by_user_group, (lambda do |ids|
     joins(:user_groups).where(user_groups: { id: ids })
   end)
+  scope :by_user_group_unique_ids, (lambda do |unique_ids|
```

Comment:
I'd like to remove this scope because I feel this is something very specific for filters. Instead, create a method: 
```ruby
def filter_with_groups(users, filters)
  return users unless filters['user_group_ids'].present?

  users.joins(:user_groups).where(user_groups: { unique_id: filters['user_group_ids'] })
end
```

Reply:
Okay